k8s-install/config.yaml

```diff
@@ -0,0 +1,34 @@
+# This ConfigMap can be used to configure a self-hosted Canal installation.
+# See `canal.yaml` for an example of a Canal deployment which uses
+# the config in this ConfigMap.
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: canal-config 
+  namespace: kube-system
+data:
+  # Configure this with the location of your etcd cluster.
+  # This must also be configured in the cni_network_config below.
+  etcd_endpoints: "http://127.0.0.1:2379"
+
+  # The interface used by flannel for host <-> host communication.
+  flannel_iface: "eth1"
+
+  # The CNI network configuration to install on each node.  Make sure
+  # this config matches your etcd / kubernetes configuration.
+  cni_network_config: |-
+    {
+        "name": "canal",
+        "type": "flannel",
+        "delegate": {
+          "type": "calico",
+          "etcd_endpoints": "http://127.0.0.1:2379",
+          "log_level": "info",
+          "policy": {
+              "type": "k8s"
+          },
+          "kubernetes": {
+              "kubeconfig": "/etc/kubernetes/worker-kubeconfig.yaml"
```

Comment:
See https://github.com/projectcalico/calico-cni/pull/177, which will add an option for making this "magic".

Until then, this isn't super portable and needs to be tweaked per-install.